### DISCUSSION TITLE
ESP32 C3 Super mini can't use PWM 1Hz error E (810) ledc: freq_hz=1 duty_resolution=0 ?

### DISCUSSION THREAD

**Discussion by Witawat** [micropython/micropython#16434] [ESP32] [answered]
i try code PWM mode 

```
frequency = 1 # PWM 1Hz
pwm = PWM(Pin(pwm_pin), freq=frequency, duty=0)
```

On runing error 

> Type 'help()' (without the quotes) then press ENTER.
> E (810) ledc: freq_hz=1 duty_resolution=0
> Traceback (most recent call last):
>   File "main.py", line 46, in <module>
> OSError: (-258, 'ESP_ERR_INVALID_ARG')
> MicroPython v1.24.1 on 2024-11-29; ESP32C3 module with ESP32C3

**Comment by robert-hh** [comment, marked as answer]
Each port has a different lowest PWM frequency. For ESP32C3 it's 5Hz. For 1Hz you may use the Timer mechanism.
